doc/developer/platform/formalism.md

```diff
@@ -194,3 +199,111 @@ A view introduces *constraints* on the capabilities and frontiers of its input a
     Outputs may need to be recovered in the case of failure, but also in other less dramatic query migration scenarios.
 
 These constraints are effected using the capability abstractions.
+The Compute layer acquires and maintains read capabilities for collections managed by it, and by the Storage layer.
+
+# Adapter
+
+The adapter layer translates SQL statements into commands for the Storage and Compute layers.
+
+The most significant differences between SQL statements and commands to the Storage and Compute layers are:
+1.  The absence of explicit timestamps in SQL statements.
+
+    A `SELECT` statement does not indicate *when* it should be run, or against which version of its input data.
+    The Adapter layer introduces timestamps to these commands, in a way that provides the appearance of sequential execution.
+
+2.  The use of user-defined and reused names in SQL statements rather than `GlobalId` identifiers.
+
+    SQL statements reference user-assigned names that may be rebound over time.
+    The meaning, and even the validity, of a query depends on the current association of user-defined name to `GlobalId`.
+    The Adapter layer maintains a map from user-assigned names to `GlobalId` identifiers, and introduces new ones as appropriate.
+
+Generally, SQL is "more ambiguous" than the Storage and Compute layers, and the Adapter layer must resolve that ambiguity.
+
+The SQL commands are a mix of DDL (data definition language) and DML (data manipulation language).
+In Materialize today, the DML commands are timestamped, and the DDL commands are largely not (although perhaps they should be).
```

Comment:
Ehhhh, idk, I think it's the opposite: the DDL commands _are_ largely timestamped! The `GlobalId::User` is the timestamp. Names and logical compaction windows can be edited after the fact; otherwise I _think_ that the Global ID forms a total order.

Reply:
So, there's a concept of "timestamp" running throughout the doc that I think does not apply to DDL statements. The `GlobalId` is useful, though we concluded I think not necessarily a total order (with independent `System`, `User`, and `Temporary` variants).  The intent, and I can try and rephrase, is that we do not have a notion of "the system as a whole at `time`" so much as "the views that happened to be installed now, each of which can be inspected at `time`".

Reply:
I think we could think of "the pTVC that is the catalog contents varying with the ambient 'system time'" but I don't know nearly enough to write that. But it could be clarifying about how to think about it. At the very least the map from `GlobalId` to other things seems like it could profitably be viewed as a `pTVC`.

Reply:
> I think we could think of "the pTVC that is the catalog contents varying with the ambient 'system time'" but I don't know nearly enough to write that. But it could be clarifying about how to think about it. At the very least the map from `GlobalId` to other things seems like it could profitably be viewed as a `pTVC`.

Yes, precisely!